File: lib/java/src/test/java/org/thryft/protocol/test/StringMapProtocolTest.java
package org.thryft.protocol.test;

import static org.junit.Assert.assertEquals;

import org.apache.thrift.TBase;
import org.apache.thrift.protocol.TProtocol;
import org.thryft.protocol.StringMapProtocol;

import com.google.common.collect.ImmutableMap;

public class StringMapProtocolTest extends ProtocolTest {
    @Override
    protected void _test(final TBase<?, ?> expected) throws Exception {
        final StringMapProtocol oprot = new StringMapProtocol();
        expected.write(oprot);
        final ImmutableMap<String, String> ostringMap = oprot.toStringMap();

        final StringMapProtocol iprot = new StringMapProtocol(ostringMap);
        final TBase<?, ?> actual = expected.getClass()
                .getConstructor(TProtocol.class).newInstance(iprot);
        assertEquals(expected, actual);
    }
}

File: lib/py/test/thryft_test/protocol/test/string_map_protocol_test.py
from thryft.protocol.string_map_protocol import StringMapProtocol
from thryft_test.protocol.test._protocol_test import _ProtocolTest


class StringMapProtocolTest(_ProtocolTest):
    def _test(self, in_object):
        oprot = StringMapProtocol()
        in_object.write(oprot)
        string_map = oprot.to_string_map()
        self.assertTrue(isinstance(string_map, dict))
        # print string_map
